contracts/YieldStrategyLever.sol

```diff
@@ -0,0 +1,492 @@
+            seriesId,
+            msg.sender,
+            amountToInvest,
+            borrowAmount
+        );
+    }
+
+    /// @notice Divest, either before or after maturity.
+    /// @param vaultId The vault to divest from.
+    /// @param seriesId The series to divest from.
+    /// @param ink The amount of collateral to recover.
+    /// @param art The amount of debt to repay.
+    /// @param minBaseOut Used to minimize slippage. The transaction will revert
+    ///     if we don't obtain at least this much of the base asset.
+    function divest(
+        bytes6 ilkId,
+        bytes12 vaultId,
+        bytes6 seriesId,
+        uint256 ink,
+        uint256 art,
+        uint256 minBaseOut
+    ) external {
+        // Test that the caller is the owner of the vault.
+        // This is important as we will take the vault from the user.
+        require(CAULDRON.vaults(vaultId).owner == msg.sender);
+
+        // Give the vault to the contract
+        giver.seize(vaultId, address(this));
+
+        IPool pool = IPool(LADLE.pools(seriesId));
+
+        // Check if we're pre or post maturity.
+        bool success;
+        if (uint32(block.timestamp) < CAULDRON.series(seriesId).maturity) {
+            IFYToken fyToken = pool.fyToken();
+            // Repay:
+            // Series is not past maturity.
+            // Borrow to repay debt, move directly to the pool.
+            bytes memory data = bytes.concat(
+                bytes1(bytes1(uint8(uint256(Operation.REPAY)))), // [0:1]
+                seriesId, // [1:7]
+                vaultId, // [7:19]
+                ilkId, // [19:25]
+                bytes32(ink), // [25:57]
+                bytes32(art) // [57:89]
+            );
+            success = IERC3156FlashLender(address(fyToken)).flashLoan(
+                this, // Loan Receiver
+                address(fyToken), // Loan Token
+                art, // Loan Amount: borrow exactly the debt to repay.
+                data
+            );
+        } else {
+            FlashJoin join = FlashJoin(
+                address(LADLE.joins(seriesId & ASSET_ID_MASK))
+            );
+            IERC20 baseAsset = IERC20(pool.base());
+
+            // Close:
+            // Series is past maturity, borrow and move directly to collateral pool.
+            bytes memory data = bytes.concat(
+                bytes1(bytes1(uint8(uint256(Operation.CLOSE)))), // [0:1]
+                seriesId, // [1:7]
+                vaultId, // [7:19]
+                ilkId, // [19:25]
+                bytes32(ink), // [25:57]
+                bytes32(art) // [57:89]
+            );
+            // We have a debt in terms of fyToken, but should pay back in base.
+            uint128 base = CAULDRON.debtToBase(seriesId, art.u128());
+            success = join.flashLoan(
+                this, // Loan Receiver
+                address(baseAsset), // Loan Token
+                base, // Loan Amount
+                data
+            );
```

Comment:
If you want to use unconventional code structures you can do this and get rid of `success`:
```
join.flashLoan(
    this, // Loan Receiver
    address(baseAsset), // Loan Token
    base, // Loan Amount
    data
) || revert FlashLoanFailure(); // revert if `flashLoan` returns `false`
```